test/onnx/gen_onnx.py

```diff
@@ -2035,6 +2035,24 @@ def gather_test():
     return ([node], [x, i], [y])
 
 
+@onnx_test()
+def gather_dyn_test():
```

Comment:
Can you add an onnx test where indices are scalar ?

Reply:
I added test `gather_scalar_test` that reflects current behavior, but I'd like some feedback on whether that behavior is what we want.

Is there any added value in creating both static and dynamic versions of this test?

Reply:
> Is there any added value in creating both static and dynamic versions of this test?
Better test coverage. But yeah, it is not much helpful in this PR since it was already covered by other tests. Good to have it anyways.